treeherder/auth/backends.py

```diff
@@ -99,32 +84,31 @@ def authenticate(self, auth_header=None, host=None, port=None):
 
         if result["status"] != "auth-success":
             logger.warning("Error logging in: {}".format(result["message"]))
-            raise TaskclusterAuthenticationFailed(result["message"])
-
-        # TODO: remove this size limit when we upgrade to django 1.10
-        # in Bug 1311967
-        email = self._get_email(result)
-        if len(email) <= 30:
-            username = email
-        else:
-            username = base64.urlsafe_b64encode(
-                hashlib.sha1(smart_bytes(email)).digest()
-                ).rstrip(b'=')[-30:]
+            raise TaskclusterAuthenticationFailedException(result["message"])
+
+        client_id = result["clientId"]
+        email = self._get_email_from_clientid(client_id)
+        username = client_id
 
         try:
-            # Find the user by their email.
-            user = self._get_user(email)
-            user.username = username
+            if email and scope_match(result["scopes"],
+                                     [["assume:mozilla-user:{}".format(email)]]):
```

Comment:
This will happen in the "email login" case.  You only get that scope if you do an LDAP login.  And even then, perhaps it's possible you don't have the scope, I think.  It can be a "*" in the TC admin case, as you said.  But the ``scope_match`` function handles the "*" as a True.